src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -386,17 +385,22 @@ public String toString() {
         return timetables.keySet();
     }
 
+    /**
+     * Add the patterns to the stop index, only if they come from a modified pattern
+     */
     private void addPatternToIndex(TripPattern tripPattern) {
-        for (Stop stop: tripPattern.getStops()) {
-            patternsForStop.put(stop, tripPattern);
+        if (tripPattern.getOriginalTripPattern() != null) {
```

Comment:
Add a method to the `TripPattern.createdByRealTimeUpdate() : booleean` and use it here.